src/ripple/app/misc/NegativeUNLVote.cpp

```diff
@@ -135,29 +112,28 @@ NegativeUNLVote::addTx(
     }
 }
 
-NodeID
+PublicKey
 NegativeUNLVote::choose(
     uint256 const& randomPadData,
-    std::vector<NodeID> const& candidates)
+    std::vector<PublicKey> const& candidates)
 {
     assert(!candidates.empty());
-    static_assert(NodeID::bytes <= uint256::bytes);
-    NodeID randomPad = NodeID::fromVoid(randomPadData.data());
-    NodeID txNodeID = candidates[0];
+    PublicKey chosenValPubKey = candidates[0];
     for (int j = 1; j < candidates.size(); ++j)
     {
-        if ((candidates[j] ^ randomPad) < (txNodeID ^ randomPad))
+        if ((*candidates[j].data() ^ *randomPadData.data()) <
```

Comment:
@nbougalis  Is this a good way to compare the candidate validators? I'm not sure if `*candidates[j].data()` is the best way to extract the data in the public key.